pir/cpp/database.cpp

```diff
@@ -142,11 +188,10 @@ StatusOr<Ciphertext> PIRDatabase::multiply(
         "Selection vector size does not match dimensions");
   }
 
-  auto database_it = db_.begin();
   try {
-    return multiply_dims(*(context_->Evaluator()), dimensions,
-                         selection_vector.begin(), selection_vector.end(),
-                         database_it, db_.end(), relin_keys, decryptor, 0);
+    DatabaseMultiplier dbm(db_, selection_vector, *(context_->Evaluator()),
```

Comment:
Why not passing the Evaluator as a shared_ptr?

Reply:
Good call. Done.